src/main/java/fr/uge/chargepointconfiguration/entities/Chargepoint.java

```diff
@@ -59,6 +62,50 @@ public class Chargepoint {
   @JoinColumn(name = "id_firmware", referencedColumnName = "id_firmware", nullable = false)
   private Firmware firmware;
 
+  /**
+   * Chargepoint's constructor with defaults values.
+   *
+   * @param serialNumberChargepoint String.
+   * @param type String.
+   * @param constructor String.
+   * @param clientId String.
+   * @param serverAddress String.
+   * @param configuration String.
+   * @param status Status.
+   */
+  public Chargepoint(String serialNumberChargepoint,
+                     String type,
+                     String constructor,
+                     String clientId,
+                     String serverAddress,
+                     String configuration,
+                     Status status,
+                     Firmware firmware) {
+    Objects.requireNonNull(serialNumberChargepoint);
+    Objects.requireNonNull(type);
+    Objects.requireNonNull(constructor);
+    Objects.requireNonNull(clientId);
+    Objects.requireNonNull(serverAddress);
+    Objects.requireNonNull(configuration);
+    Objects.requireNonNull(status);
+    Objects.requireNonNull(firmware);
+    this.serialNumberChargepoint = serialNumberChargepoint;
+    this.type = type;
+    this.constructor = constructor;
+    this.clientId = clientId;
+    this.serverAddress = serverAddress;
+    this.configuration = configuration;
+    this.status = status;
+    this.firmware = firmware;
+  }
```

Comment:
If you want to have less lines, you could do for example :
this.type = Objects.requireNonNull(type);